sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -322,6 +337,64 @@ namespace Azure { namespace Storage { namespace Blobs {
         std::move(result), std::move(commitBlockListResponse.RawResponse));
   }
 
+  Azure::Response<Models::UploadBlockBlobFromUriResult> BlockBlobClient::UploadFromUri(
+      const std::string& sourceUri,
+      const UploadBlockBlobFromUriOptions& options,
+      const Azure::Core::Context& context) const
+  {
+    _detail::BlockBlobClient::UploadBlockBlobFromUriOptions protocolLayerOptions;
+    protocolLayerOptions.CopySource = sourceUri;
+    protocolLayerOptions.CopySourceBlobProperties = options.CopySourceBlobProperties;
+    protocolLayerOptions.BlobContentType = options.HttpHeaders.ContentType;
+    protocolLayerOptions.BlobContentEncoding = options.HttpHeaders.ContentEncoding;
+    protocolLayerOptions.BlobContentLanguage = options.HttpHeaders.ContentLanguage;
+    protocolLayerOptions.BlobContentMD5 = options.HttpHeaders.ContentHash.Value;
+    protocolLayerOptions.BlobCacheControl = options.HttpHeaders.CacheControl;
+    protocolLayerOptions.BlobContentDisposition = options.HttpHeaders.ContentDisposition;
+    protocolLayerOptions.Metadata
+        = std::map<std::string, std::string>(options.Metadata.begin(), options.Metadata.end());
+    protocolLayerOptions.BlobTagsString = std::accumulate(
+        options.Tags.begin(),
+        options.Tags.end(),
+        std::string(),
+        [](const std::string& a, const std::pair<std::string, std::string>& b) {
+          return a + (a.empty() ? "" : "&") + _internal::UrlEncodeQueryParameter(b.first) + "="
+              + _internal::UrlEncodeQueryParameter(b.second);
+        });
```

Comment:
I saw this same callback before in the PR.
Consider defining the function somewhere and using it as arg here instead of implementing twice.